### PR TITLE
std: Update crate docs

### PR DESCRIPTION
**Pull request by brson**
Attempted to organize them in a way more relevant to what newbies
would be interested in hearing.

I am not satisfied by this at all, but by virtue of deleting old links alone I think it is an improvement.

r? @steveklabnik

### CODE REVIEW

File: src/libstd/lib.rs [hunk 1]
```diff
@@ -12,33 +12,35 @@
 //!
 //! The Rust Standard Library provides the essential runtime
 //! functionality for building portable Rust software.
-//! It is linked to all Rust crates by default.
 //!
-//! ## Intrinsic types and operations
+//! It is linked to all Rust crates by default as though they
```
Comment (echochamber) [inline src/libstd/lib.rs]: > The rust standard library is available to all rust crates by default, just as if... 

Small wording suggestion that I think makes this more newbie friendly :)

Also, as a newbie, I don't quite understand what "crate-level" means in this context? As far as my understanding goes, there's no way to automatically have `extern crate some_crate` for all modules in a crate (besides the std library), but its very possible I am mistaken.

File: src/libstd/lib.rs [hunk 2]
```diff
@@ -56,42 +58,36 @@
 //! macro, and for converting from strings use the
 //! [`FromStr`](str/trait.FromStr.html) trait.
 //!
-//! ## Platform abstractions
+//! Data may be shared by placing it a reference-counted box, the
```
Comment (echochamber) [inline src/libstd/lib.rs]: > Data may be shared by placing it **in** a reference-counted box

File: src/libstd/lib.rs [hunk 3]
```diff
@@ -56,42 +58,36 @@
...
+//! Besides basic data types, the standard library is largely concerned
+//! with abstracting over differences in common platforms, most notably
+//! Windows and Unix derivatives.
 //!
-//! ## The Rust prelude and macros
+//! Common types of I/O, including [files](fs/struct.File.html),
+//! [TCP](net/struct.TcpStream.html),
+//! [UDP](net/struct.UdpSocket.html), are defined in the
+//! [`io`](io/index.html), [`fs`](fs/index.html), and
+//! [`net`](net/index.html) modulesu.
```
Comment (echochamber) [inline src/libstd/lib.rs]: > modulesu

modules

File: src/libstd/lib.rs [hunk 4]
```diff
@@ -56,42 +58,36 @@
...
 //!
-//! Finally, the [`prelude`](prelude/index.html) defines a
-//! common set of traits, types, and functions that are made available
-//! to all code by default. [`macros`](macros/index.html) contains
-//! all the standard macros, such as `assert!`, `panic!`, `println!`,
-//! and `format!`, also available to all Rust code.
+//! The [`thread`](thread/index.html) module contains Rust's threading
+//! abstractions.  [`sync`](sync/index.html) contains further,
+//! primitive, shared memory types, including
```
Comment (echochamber) [inline src/libstd/lib.rs]: >//! The \[\`thread\`\](thread/index.html) module contains Rust's threading
>//! abstractions.  \[\`sync\`\](sync/index.html) contains further,
>//! primitive, shared memory types, including

//! The \[\`thread\`\](thread/index.html) module contains Rust's threading
//! abstractions.  \[\`sync\`\](sync/index.html) contains further
//! primitive shared memory types, including

**Comment by echochamber** [comment]
Hey, was originally just going to weigh in my perspective as a rust newbie on your changes, but ended up doing a proofread also. Sorry if my comments are pedantic!

File: src/libstd/lib.rs [hunk 5]
```diff
@@ -12,33 +12,35 @@
...
 //! The Rust Standard Library provides the essential runtime
 //! functionality for building portable Rust software.
-//! It is linked to all Rust crates by default.
 //!
-//! ## Intrinsic types and operations
+//! It is linked to all Rust crates by default as though they
+//! contained a crate-level `extern crate std` crate import. Therefore
+//! the standard library can be accessed in `use` statements through
+//! the path `std`, as in `use std::thread`, or in expressions through
+//! the absolute path `::std`, as in `::std::thread::sleep_ms(100)`.
```
Comment (echochamber) [inline src/libstd/lib.rs]: Is it correct to refer to ::std as a path or a namespace or are both correct?

Comment (steveklabnik) [inline src/libstd/lib.rs]: both are, i'd say

File: src/libstd/lib.rs (hunk 1 above)
Comment (steveklabnik) [inline src/libstd/lib.rs]: It means 'in your crate root', which is how we usually say it. i'd update it to that wording.

**Comment by steveklabnik** [comment]
After a few nits, r=me. I've also thought this could be improved but was never really satisfied with how to do so.

**Pull request merged**